app/gateway/database/postgres/account.go

```diff
@@ -0,0 +1,169 @@
+package postgres
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"stonehenge/app/core/model/account"
+	"stonehenge/app/core/types/currency"
+	"stonehenge/app/core/types/document"
+	"stonehenge/app/core/types/id"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+type accountRepo struct {
+	tx Transaction
+	db *pgxpool.Pool
+}
+
+func (t *accountRepo) GetWithCPF(ctx context.Context, document document.Document) (*account.Account, error) {
+	const query string = "select * from accounts where document = $1"
+	ret := t.db.QueryRow(ctx, query, document)
+	acc, err := parseAccount(ret)
+	if err != nil {
+		return nil, account.ErrNotFound
+	}
+	return acc, nil
+}
+
+func (t *accountRepo) StartOperation(ctx context.Context) (context.Context, error) {
+	return t.tx.Begin(ctx)
```

Comment:
você realizou algum teste concorrente? to achando que pode dar conflito entre várias transactions

Reply:
Ainda não testei, mas pensei que ele criasse um objeto transaction pra cada chamada. Vou analisar isso. Valeu pelo toque